### ISSUE TITLE
Control Endpoint callbacks

### ISSUE THREAD

**Issue by kuldeepdhaka** [libopencm3/libopencm3#449]
issue: we should clear the EP_DIR bit since control endpoint do not have direction. (its bidirectional) and set the internal control ep callback functions (that means throw the user passed callback).
https://github.com/libopencm3/libopencm3/blob/master/lib/usb/usb_f103.c#L123

**Comment by kuldeepdhaka** [comment]
moved common-issue1 to another issue thread.

**Comment by kuldeepdhaka** [comment]
since common-issue1 is not going to be implemented, implementing this is not required anymore since, we are assuming that their is only one control endpoint and we are not going to mess with its callback (only once set in init).
